projects/components/src/table/controls/table-controls-api.ts

```diff
@@ -1,58 +1,77 @@
 import { Dictionary } from '@hypertrace/common';
-import { SelectOption } from '../../select/select-option';
+import { FilterOperator } from '../../filtering/filter/filter-operators';
 import { TableFilter } from '../table-api';
 
-export interface SelectControl {
-  placeholder?: string;
-  default?: SelectOption<TableControlOption>;
-  options: SelectOption<TableControlOption>[];
+export const enum TableControlOptionType {
+  Filter = 'filter',
+  Property = 'property',
+  Unset = 'unset'
 }
 
-export interface SelectChange {
-  select: SelectControl;
-  value: TableControlOption;
-}
+export type TableControlOption = TableUnsetControlOption | TableFilterControlOption | TablePropertyControlOption;
 
-export interface CheckboxControl {
+export interface TableFilterControlOption {
+  type: TableControlOptionType.Filter;
   label: string;
-  value: boolean;
-  options: TableCheckboxOptions;
+  metaValue: TableFilter;
 }
 
-export interface CheckboxChange {
-  checkbox: CheckboxControl;
-  option: TableControlOption<boolean>;
+export interface TableUnsetControlOption {
+  type: TableControlOptionType.Unset;
+  label: string;
+  metaValue: string;
```

Comment:
what is meta value supposed to do? How is it different than 'value' ?

Reply:
Please see [this](https://github.com/hypertrace/hypertrace-ui/pull/823/files#r626981848) comment below.